src/app/GitUI/Editor/Diff/DiffHighlightService.cs

```diff
@@ -230,114 +232,141 @@ private void SetText(ref string text)
     private static void MarkInlineDifferences(IDocument document, IReadOnlyList<ISegment> linesRemoved, IReadOnlyList<ISegment> linesAdded, int beginOffset)
     {
         Func<ISegment, string> getText = line => document.GetText(line.Offset + beginOffset, line.Length - beginOffset);
-        document.MarkerStrategy.AddMarkers(GetDifferenceMarkers(document.GetCharAt, getText, linesRemoved, linesAdded, beginOffset));
+        document.MarkerStrategy.AddMarkers(GetDifferenceMarkers(getText, linesRemoved, linesAdded, beginOffset));
     }
 
-    private static IEnumerable<TextMarker> GetDifferenceMarkers(Func<int, char> getCharAt, Func<ISegment, string> getText, IReadOnlyList<ISegment> linesRemoved, IReadOnlyList<ISegment> linesAdded, int beginOffset)
+    private static IEnumerable<TextMarker> GetDifferenceMarkers(Func<ISegment, string> getText, IReadOnlyList<ISegment> linesRemoved, IReadOnlyList<ISegment> linesAdded, int beginOffset)
     {
         foreach ((ISegment lineRemoved, ISegment lineAdded) in LinesMatcher.FindLinePairs(getText, linesRemoved, linesAdded))
         {
-            foreach (TextMarker marker in GetDifferenceMarkers(getCharAt, lineRemoved, lineAdded, beginOffset))
+            foreach (TextMarker marker in GetDifferenceMarkers(getText, lineRemoved, lineAdded, beginOffset))
             {
                 yield return marker;
             }
         }
     }
 
-    internal static IEnumerable<TextMarker> GetDifferenceMarkers(Func<int, char> getCharAt, ISegment lineRemoved, ISegment lineAdded, int lineStartOffset)
+    internal static IReadOnlyList<TextMarker> GetDifferenceMarkers(Func<ISegment, string> getText, ISegment lineRemoved, ISegment lineAdded, int beginOffset)
     {
-        int lineRemovedEndOffset = lineRemoved.Length;
-        int lineAddedEndOffset = lineAdded.Length;
-        int endOffsetMin = Math.Min(lineRemovedEndOffset, lineAddedEndOffset);
-        int beginOffset = lineStartOffset;
-        int reverseOffset = 0;
-
-        while (beginOffset < endOffsetMin)
+        string textRemoved = getText(lineRemoved);
```

Comment:
This should be using Span instead, but that is harder with subfunctions.
With #11947 all these allocations are only done when a diff is first added, not every time something is marked in the document,
The processing is only visible for very large documents.

Reply:
```
Func<ISegment, string> getText = line => document.GetText(line.Offset + beginOffset, line.Length - beginOffset);
```
was added in 3e395331ee9431d871b2961dd371bb8cca4cfe82.
In order to use `Span`, the interface `IDocument` of `ICSharpCode.TextEditor` would need to be extended.
But in the follow-up, you have moved the access to `IDocument` up in the callstack.
Anyway, since `Span` cannot be used in lambda functions nor in iterator functions, `getText` still needs to return `string`, and we must create `string` instances for calling `GetWords` / `FindBestMatch`.
So, we can make only limited use of `ReadOnlySpan`: 44342c3. The gain might be small or even negative.